### ISSUE TITLE
Server crash

### ISSUE THREAD

**Issue by Ancienty** [DecentSoftware-eu/DecentHolograms#148]
### Just making sure

- [x] I made sure, this bug is not already known or reported.
- [X] I believe, that this bug is fixable.

### Reproduction

No idea.

### Solution

Hello,

So yesterday my server crashed and I opened an issue tracker on PandaSpigot's GitHub, they said that DecentHolograms caused a deadlock and somehow crashed the server and told me to open an issue tracker here.

I'm not entirely sure if this crash happened because of DecentHolograms but here I am.
Sorry in advance if I shouldn't have reported this.

The watchdog reports:
[log.txt](https://github.com/DecentSoftware-eu/DecentHolograms/files/12028777/log.txt)

### Server Version

PandaSpigot 1.8.9

### Client Version

1.8.9

### Plugin Version

2.7.12

### Log

Log included as .txt file above

**Comment by Andre601** [comment]
Please share the logs through something like paste.helpch.at

Not everyone here likes to download txt files to read the logs.

**Comment by Ancienty** [comment]
https://paste.helpch.at/ilumacozup.makefile

Here you go.

**Comment by Andre601** [comment]
Thanks. Mind also sharing your full log from server start for better context?

**Comment by Ancienty** [comment]
I actually can't, deleted that part of the log while I was uploading the file here. :(
Although I don't really remember anything being out of the ordinary, no error logs or anything from DecentHolograms or any other plugins.

The crash just happened out of nowhere.

**Comment by Ancienty** [comment]
any updates?

**Comment by Andre601** [comment]
Not without a proper server log as that gives us the necessary context to understand the situation.

**Comment by Ancienty** [comment]
Will send it again if the crash re-occurs then.

**Comment by Ancienty** [comment]
Crashed again but paste.helpch.at isn't working rn, I can't share the entire logs.
If you're okay with me sending it as a .txt file, let me know.

**Comment by Andre601** [comment]
There is also paste.gg, mclo.gs or gist.github.com

**Comment by Ancienty** [comment]
https://mclo.gs/E7U6NsW

Please don't mind the errors thrown by Essentials, they've also been reported.

**Comment by Andre601** [comment]
The only thing I can find that is remotely related to DH is an error that seems to be related to the Javascript expansion of PlaceholderAPI, therefore not a DH issue and for sure not a cause for a crash I would assume.

One thing that confuses me a lot here is, that your initial logs have "CombatLog v1.8.11-**hack**" in their lines, which sounds a lot like a leaked resource...

If that's the case, then please be honest about it... and also don't use leaked resources.

Either way, this is not a DH issue and it's for sure not the cause of your server crash.

**Comment by Ancienty** [comment]
CombatLog is a free resource lol why would I use a leaked version of it. I had never noticed that name though, thanks for stating that out. I'll try re-installing it nevertheless.

I also thought it wasn't related to DH but I was sent here to get your thoughts, sorry for wasting your time.
Thank you for the quick replies.

**Comment by Andre601** [comment]
Your first log mentions vulcan as one of the first in the stacktraces, so maybe it's that anticheat (iirc it was an anticheat?)

I think you have more chances for help by asking on a discord server like [HelpChat](https://discord.gg/HelpChat) which are more focused on general plugin/Server help.

**Comment by Ancienty** [comment]
Okay I'll try my luck there then
Thank you.

**Comment by qlnn** [comment]
skill issues

**Comment by SladeHazard** [comment]
u are literally using a plugin from 2013, relating this to decentholograms is just funny

![image](https://github.com/DecentSoftware-eu/DecentHolograms/assets/45261094/bfb35987-65f5-44c3-94bb-72e3db5c9cb8)

**Comment by SladeHazard** [comment]
one or two blocked threads dont cause crashes,

Caused by: java.lang.NoSuchMethodError: org/bukkit/Server.getOnlinePlayers()[Lorg/bukkit/entity/Player; (loaded from file:/root/Skyblock/cache/patched_1.8.8.jar by sun.misc.Launcher$AppClassLoader@9a664ecd) called from class com.jackproehl.plugins.Metrics (loaded from file:/root/Skyblock/plugins/CombatLog-1.8.11-fix.jar by org.bukkit.plugin.java.PluginClassLoader@bd2b491c).

**Comment by Andre601** [comment]
> u are literally using a plugin from 2013, btw what version is ur java ? + the company
> 
> ![image](https://user-images.githubusercontent.com/45261094/270061779-bfb35987-65f5-44c3-94bb-72e3db5c9cb8.png)

Just in case this shows "DecentHolograms" (Quotes are on purpose) on dev.bukkit.org/Curseforge:
**This is NOT an official release by d0by or any of the DH maintainers/Contributers! Only official downloads exist on Spigot and the dev.bukkit.org/Curseforge page should be reported for license/copyright violation as it is a redistribution without proper notice of original author!**

**Comment by Ancienty** [comment]
This doesn't occur any more, but I'd like to reply to some of the messages you guys sent:

Clearly you haven't read the OP, I've said I've been redirected here by PandaSpigot, their dev thought it could be related to DecentHolograms, I did not. I also stated "I'm not entirely sure if this crash happened because of DecentHolograms but here I am.".

This was in fact actually caused by PandaSpigot which after a few releases was fixed.

I'm still using a plugin that was developed back in 2013, true, and my server's version is 1.8, which came out in 2014, so it's not that big of a deal. The plugin does nothing but checking if the players have been in combat in the last few seconds, it's not that complicated and does not require constant updates.

I even forgot I had posted this issue, saw it through my mails and came here to write this.
IDK how to use github, feel free to tag this as closed/solved.
